e2e/ansible/playbooks/feature/ha/ha.yml

```diff
@@ -167,71 +142,92 @@
+           pod_restarts: "{{ result.stdout.split()[3] }}"
            pod_ip: "{{ result.stdout.split()[5] }}"
 
        - name: Verify percona is running w/o restarts
-         debug: 
+         debug:
            msg: "Percona pod health-check is successful"
          failed_when: "pod_status != 'Running' and pod_restarts != '0'"
 
        - name: Create a test db on percona to confirm health
-         shell: mysql -uroot -pk8sDem0 -h {{pod_ip}} -e "CREATE DATABASE Inventory;"
-         args: 
+         shell: |
+           sleep 100
+           mysql -uroot -pk8sDem0 -h {{pod_ip}} -e "CREATE DATABASE Inventory;"
+         args:
            executable: /bin/bash
          delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
-         register: result 
+         register: result
          failed_when: "result.rc != 0"
 
        ########################################################
        # REVERT FAULTS AND CLEANUP                            #
        # (Includes commands, delete applications etc.,)       #
-       ########################################################     
+       ########################################################
 
        - name: Uncordon the K8S node as part of cleanup
          shell: source ~/.profile; kubectl uncordon {{ctrl_node}}
          args:
            executable: /bin/bash
          register: result
          delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
-   
-       - include: ha-cleanup.yml
 
-       - name: Terminate the log aggregator
-         shell: source ~/.profile; killall stern
-         args:
-           executable: /bin/bash
-         delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
 
-       - set_fact: 
+       - name: Test Passed
+         set_fact:
            flag: "Pass"
 
-     rescue:  
-       - set_fact: 
+     rescue:
+       - name: Test Failed
+         set_fact:
            flag: "Fail"
- 
-     always: 
-       - name: Send slack notification
-         slack:
-           token: "{{ lookup('env','SLACK_TOKEN') }}"
-           msg: '{{ ansible_date_time.time }} TEST: {{test_name}}, RESULT: {{ flag }}'
-         when: slack_notify | bool and lookup('env','SLACK_TOKEN') 
-
-           
-         
-       
+
+     always:
+       - block:
+
+
+           - include: ha-cleanup.yml
+
+           - name: Test Cleanup Passed
+             set_fact:
+               cflag: cleanup_pass
+
+         rescue:
+           - name: Test Cleanup Failed
+             set_fact:
+               cflag: cleanup_fail
+
+
+
+         always:
+
+           - name: Terminate the log aggregator
+             shell: source ~/.profile; killall stern
+             args:
+               exedcutable: /bin/bash
+             delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+
+           - name: Send slack notification
+             slack:
+              token: "{{ lookup('env','SLACK_TOKEN') }}"
+              msg: '{{ ansible_date_time.time }} TEST: {{test_name}}, RESULT: {{ flag }}'
```

Comment:
Result should give both test and cleanup status.

Ex: RESULT: {{ flag }}, {{cfgla}}